source/index.ts

```diff
@@ -0,0 +1,744 @@
+	*/
+	onDidChange(key: string, callback: (...args: unknown[]) => void): () => unknown {
+		if (typeof key !== 'string') {
+			throw new TypeError(`Expected \`key\` to be of type \`string\`, got ${typeof key}`);
+		}
+
+		if (typeof callback !== 'function') {
+			throw new TypeError(`Expected \`callback\` to be of type \`function\`, got ${typeof callback}`);
+		}
+
+		const getter: () => unknown = () => this.get(key);
+
+		return this.handleChange(getter, callback);
+	}
+
+	/**
+	Watches the whole config object, calling `callback` on any changes.
+
+	@param callback - A callback function that is called on any changes. When a `key` is first set `oldValue` will be `undefined`, and when a key is deleted `newValue` will be `undefined`.
+	*/
+	onDidAnyChange(callback: (newValue: unknown, oldValue: unknown) => void): () => unknown {
+		if (typeof callback !== 'function') {
+			throw new TypeError(`Expected \`callback\` to be of type \`function\`, got ${typeof callback}`);
+		}
+
+		const getter: () => unknown = () => this.store;
+
+		return this.handleChange(getter, callback);
+	}
+
+	handleChange(getter: () => unknown, callback: (newValue: unknown, oldValue: unknown) => void): () => unknown {
+		let currentValue = getter();
+
+		const onChange: () => unknown = () => {
+			const oldValue = currentValue;
+			const newValue = getter();
+
+			try {
+				// TODO: Use `util.isdeepEqual` when targeting Node.js 10
+				assert.deepEqual(newValue, oldValue);
+			} catch (_) {
+				currentValue = newValue;
+				callback.call(this, newValue, oldValue);
+			}
+		};
+
+		this.events?.on('change', onChange);
+		return () => this.events?.removeListener('change', onChange);
+	}
+
+	_encryptData(data: string | Buffer): string | Buffer {
+		if (!this.encryptionKey) {
+			return data;
+		}
+
+		try {
+			// Check if an initialization vector has been used to encrypt the data
+			if (data.slice(16, 17).toString() === ':') {
+				const initializationVector = data.slice(0, 16);
+				const password = crypto.pbkdf2Sync(this.encryptionKey, initializationVector.toString(), 10000, 32, 'sha512');
+				const decipher = crypto.createDecipheriv(encryptionAlgorithm, password, initializationVector);
+				const slicedData: any = data.slice(17);
+				return Buffer.concat([decipher.update(slicedData), decipher.final()]);
+			}
+
+			// Legacy decryption without initialization vector
+			const decipher = crypto.createDecipher(encryptionAlgorithm, this.encryptionKey);
+			const legacyData: any = data;
+			return Buffer.concat([decipher.update(legacyData), decipher.final()]);
+		} catch (_) {
+			return data;
+		}
+	}
+
+	createPlainObject(): {[key: string]: T} {
+		return Object.create(null);
+	}
+
+	get size(): number {
+		return Object.keys(this.store).length;
+	}
+
+	get store(): {[key: string]: T} {
+		try {
+			const data: string | Buffer = fs.readFileSync(this.path, this.encryptionKey ? null : 'utf8');
+			const dataString = this._encryptData(data) as string;
```

Comment:
You should not use `as string` here as encryptData could return a Buffer. Generally, stay away from `as` unless necessary as it could easily hide type errors.